DigitalLearningSolutions.Data.Tests/DataServices/PasswordResetDataServiceTests.cs

```diff
@@ -90,24 +111,153 @@ public async Task Does_Not_Match_Reset_Passwords_If_No_Reset_Password_With_Given
-                testDelegateUser.EmailAddress,
-                hashToCheck);
+            var matches = await service.FindMatchingResetPasswordEntitiesWithUserDetailsAsync
+            (
+                testDelegateUser.EmailAddress!,
+                HashNotYetInDb
+            );
 
             // Then
             matches.Count.Should().Be(0);
         }
+
+        [Test]
+        public async Task Removing_reset_hash_sets_ResetPasswordId_to_null_for_admin_user()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var userId = UserTestHelper.GetDefaultAdminUser().Id;
+            var userRef = new UserReference(userId, UserType.AdminUser);
+            var resetPasswordId = await GivenResetPasswordWithHashExistsForUsersAsync(HashNotYetInDb, new[] { userRef });
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var userAfterRemoval = userDataService.GetAdminUserById(userId);
+            userAfterRemoval.Should().NotBeNull();
+            userAfterRemoval?.ResetPasswordId.Should().BeNull();
+        }
+
+        [Test]
+        public async Task Removing_reset_hash_sets_ResetPasswordId_to_null_for_delegate_user()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var userId = UserTestHelper.GetDefaultDelegateUser().Id;
+            var userRef = new UserReference(userId, UserType.DelegateUser);
+            var resetPasswordId = await GivenResetPasswordWithHashExistsForUsersAsync(HashNotYetInDb, new[] { userRef });
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var userAfterRemoval = userDataService.GetDelegateUserById(userRef.Id);
+            userAfterRemoval.Should().NotBeNull();
+            userAfterRemoval?.ResetPasswordId.Should().BeNull();
+        }
+
+        [Test]
+        public async Task Removing_reset_hash_from_admin_user_removes_ResetPassword_entity()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var userId = UserTestHelper.GetDefaultAdminUser().Id;
+            var userRef = new UserReference(userId, UserType.AdminUser);
+            var resetPasswordId = await GivenResetPasswordWithHashExistsForUsersAsync(HashNotYetInDb, new[] { userRef });
+
+            // When
+            await service.RemoveResetPasswordAsync(resetPasswordId);
+
+            // Then
+            var matchingResetPasswords = connection.Query<ResetPassword>
+            (
+                "SELECT * FROM ResetPassword WHERE ID = @ResetPasswordId",
+                new { ResetPasswordId = resetPasswordId }
+            );

```

Comment:
There are some test helper under "Helpers" folder that have the queries that we need only for tests - maybe worth separating this query into a helper too for consistency and better readability? (or just into a private function, if you prefer)